www/class/CentreonLDAPAdmin.class.php

```diff
@@ -57,6 +57,29 @@ public function __construct($pearDB)
         $this->centreonLog = new CentreonLog();
     }
 
+    /**
+     * used to sanitize key and value in array
+     * @param array<mixed> $inputArray
+     * @return array<mixed>
+     */
+    public function sanitizeInputArray(array $inputArray): array
+    {
+        $sanitizedArray = [];
+        foreach ($inputArray as $key => $value) {
+            if ($key === 'address') {
+                $key = filter_var($key, FILTER_SANITIZE_STRING);
+                $value = filter_var($value, FILTER_SANITIZE_STRING);
+            } else {
+                $key = filter_var($key, FILTER_VALIDATE_INT);
+                $value = filter_var($value, FILTER_VALIDATE_INT);
+            }
+            if (false !== $key && false !== $value) {
```

Comment:
FILTER_SANITIZE_STRING will not return false. But may return an empty string.
Should we use an empty address ?

Reply:
updated condition at line 76